src/main/java/de/tobiasroeser/lambdatest/proxy/TestProxy.java

```diff
@@ -135,55 +135,64 @@ public static IgnoreMethod ignoreMethod(final String name) {
 				return "Proxy[" + mkString(interfaces, " & ") + "]@" + System.identityHashCode(proxy);
 			} else {
 				final String methodSignature = methodSignature(method);
-				throw new UnsupportedOperationException("" +
-						"Unhandled call: proxy=" + proxy + ", method=" + method + ", args=" +
-						(args == null ? "null" : mkString(args, ", ")) +
-						"\nTo handle this call in the proxy delegate object, add a method with the following signature:"
-						+
-						"\n  " + methodSignature + " {\n  }" +
-						"\n");
+				final String interfaceName = method.getDeclaringClass().getSimpleName();
+
+				final String argObjects = args == null ? "" : mkString(args, ", ");
+				throw new UnsupportedOperationException(""
+						+ "Unhandled call: proxy=" + proxy + ", method=" + method + ", args=" + argObjects
+						+ "\nTo handle this call in the proxy, add the following to " + interfaceName
+						+ "\n ==> " + methodSignature + "{}"
+						+ "\n");
 			}
 		});
 	}
 
 	private static String methodSignature(final Method method) {
-		final String argList = mkArgListString(method.getParameterTypes());
+		final String argList = mkArgListString(method.getGenericParameterTypes());
 		final String methodName = method.getName();
-		final String returnTypeName = method.getReturnType().getSimpleName();
-		return "public " + returnTypeName + " " + methodName + "(" + argList + ")";
+		final String returnTypeName = removeAllPackages(method.getGenericReturnType().getTypeName());
+		final TypeVariable<Method>[] typeParameters = method.getTypeParameters();
+		final String typeVariables = typeParameters == null || typeParameters.length == 0 ? "" : mkTypeVariablesList(typeParameters) + " ";
+
+		return "public " + typeVariables + returnTypeName + " " + methodName + "(" + argList + ")";
+	}
+
+	private static String filterLetters(String string) {
+		return string.replaceAll("[^a-zA-Z0-9]","");
 	}
 
-	private static String mkArgListString(Class<?>[] types) {
-		if (types == null) {
+	private static String mkTypeVariablesList(TypeVariable<Method>[] typeParameters){
+		return "<" + mkString(Arrays.stream(typeParameters).map(t -> t.getTypeName()).collect(Collectors.toList()),", ") + ">";
+	}
+	private static String mkArgListString(final Type[] args) {
+		if (args == null) {
 			return "";
 		}
+		final List<String> argsWithClassAndParameterName = new LinkedList<>();
+		for(int i =0; i < args.length; i++) {
+			final Type arg = args[i];
+			final String argWithPackages = arg.getTypeName();
+			final String className = removeAllPackages(argWithPackages);
+
+			final String argName = filterLetters(Util.decapitalize("x"));
```

Comment:
You replace the previous naming strategy (e.g. `List list1, List list2, Interger integer1`) with something less readable (`List<Integer> x0, List<String> x1, Integer x2`). I don't like that.